rasa/shared/nlu/training_data/training_data.py

```diff
@@ -712,6 +712,16 @@ def __init__(
         self.training_examples = training_examples
         self.responses = responses
 
+    @lazy_property
+    def nlu_examples(self) -> List[Message]:
+        """Filter training examples that didn't come from nlu data."""
+        return [ex for ex in self.training_examples if not ex.is_core_message()]
+
+    @lazy_property
+    def entity_examples(self) -> List[Message]:
+        """Filter nlu examples that don't contain entities."""
+        return [ex for ex in self.nlu_examples if ex.get(ENTITIES)]
```

Comment:
I think we can remove those again as `TrainingDataChunk` now inherits from `TrainingData` once you merge `partial-dataset-loading` branch into yours.